### PR TITLE
Update Gulp links

### LINKED ISSUE #12364: "Use Gulp" hyperlink issue
The URL for the "Use Gulp" hyperlinks in the first paragraph and the Additional resources section differs from the one for this page but the URL for those hyperlinks gets redirected to this page ("Use Grunt in ASP.NET Core").

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2e3b23da-ff3d-daf6-c56a-2b8ddf79f805
* Version Independent ID: 5ea39e72-fcfc-ddbe-2f4d-f4166dfbe5a8
* Content: [Use Grunt in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/client-side/using-grunt?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/client-side/using-grunt.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/client-side/using-grunt.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #12364

Updates/removes a few Gulp links following the *Retire Gulp* PR (https://github.com/aspnet/AspNetCore.Docs/pull/12360).

Thanks @BraveThrasher! :rocket:

### CODE REVIEW

File: aspnetcore/performance/performance-best-practices.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ ASP.NET Core apps with complex front-ends frequently serve many JavaScript, CSS,
 Recommendations:
 
 * **Do** use ASP.NET Core's [built-in support](xref:client-side/bundling-and-minification) for bundling and minifying client assets.
-* **Do** consider other third-party tools, such as [Gulp](xref:client-side/using-gulp) or [Webpack](https://webpack.js.org/) for complex client asset management.
+* **Do** consider other third-party tools, such as [Gulp](https://gulpjs.com/) or [Webpack](https://webpack.js.org/) for complex client asset management.
```
Comment (tdykstra) [inline aspnetcore/performance/performance-best-practices.md]:
```suggestion
* **Do** consider other third-party tools, such as [Webpack](https://webpack.js.org/), for complex client asset management.
```

File: aspnetcore/client-side/bundling-and-minification.md [hunk 2]
```diff
@@ -319,11 +319,11 @@ In this example, any tasks defined within the `MyPreCompileTarget` target run be
 ========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
 ```
 
-Alternatively, Visual Studio's Task Runner Explorer may be used to bind Gulp tasks to specific Visual Studio events. See [Running default tasks](xref:client-side/using-gulp#running-default-tasks) for instructions on doing that.
+Alternatively, Visual Studio's Task Runner Explorer may be used to bind Gulp tasks to specific Visual Studio events.
 
 ## Additional resources
 
-* [Use Gulp](xref:client-side/using-gulp)
+* [Gulp](https://gulpjs.com/)
```
Comment (tdykstra) [inline aspnetcore/client-side/bundling-and-minification.md]:
```suggestion
```

File: aspnetcore/client-side/bundling-and-minification.md [hunk 3]
```diff
@@ -319,11 +319,11 @@ In this example, any tasks defined within the `MyPreCompileTarget` target run be
 ========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
 ```
 
-Alternatively, Visual Studio's Task Runner Explorer may be used to bind Gulp tasks to specific Visual Studio events. See [Running default tasks](xref:client-side/using-gulp#running-default-tasks) for instructions on doing that.
+Alternatively, Visual Studio's Task Runner Explorer may be used to bind Gulp tasks to specific Visual Studio events.
```
Comment (tdykstra) [inline aspnetcore/client-side/bundling-and-minification.md]:
```suggestion
```

File: aspnetcore/client-side/bundling-and-minification.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ Browsers are fairly verbose with regard to HTTP request headers. The total bytes
...
+The MVC and Razor Pages project templates provide an out-of-the-box solution for bundling and minification consisting of a JSON configuration file. Third-party tools, such as the [Gulp](https://gulpjs.com/) and [Grunt](xref:client-side/using-grunt) task runners, accomplish the same tasks with a bit more complexity. A third-party tool is a great fit when your development workflow requires processing beyond bundling and minification&mdash;such as linting and image optimization. By using design-time bundling and minification, the minified files are created prior to the app's deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```
Comment (tdykstra) [inline aspnetcore/client-side/bundling-and-minification.md]:
```suggestion
The MVC and Razor Pages project templates provide an out-of-the-box solution for bundling and minification consisting of a JSON configuration file. Third-party tools, such as the [Grunt](xref:client-side/using-grunt) task runner, accomplish the same tasks with a bit more complexity. A third-party tool is a great fit when your development workflow requires processing beyond bundling and minification&mdash;such as linting and image optimization. By using design-time bundling and minification, the minified files are created prior to the app's deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```

**Comment by guardrex** [comment]
hum ... thought the commit comment would become a comment here.

Copied from the commit message ...

@tdykstra ... It's more complete reduction in Gulp than I anticipated. 

The bundling and minification topic still contains Gulp content: https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification#consume-bundleconfigjson-from-gulp.

Do you want me to remove that content? If so, I can do it on this PR.

**Comment by tdykstra** [comment]
On that one, looks like we'd need to replace Gulp with something else that does the same thing, so it's more than just link removal; I'd say we should create a separate issue for that work.

**Comment by guardrex** [comment]
Will do. Thanks.

**Pull request merged**